### PR TITLE
#160374729 Fix Left Sidebar Bug; Create an Application Shell

### PR DESCRIPTION
**Pull request by johnmutuma5**
#### What does this PR do?
-  Creates an application shell 
  - Avoid re-instantiation of the header bar, left-side navigation bar, notifications pane and the side drawer when navigating the application
  - Fixes unexpected behavior when switching the left sidebar links
  - Fix failing tests in approvals sagas

#### Description of Task to be completed?
- Create an application shell for constant components
- Fix active link transition bug in the left side navigation bar

#### How should this be manually tested?
- Clone the repo - `https://github.com/andela/travel_tool_front.git`
- Switch to the repo directory - `cd travel_tool_front`
- Checkout into this branch - `git checkout ch-app-shell-160374729`
- Install dependencies and start the app - `make start`
- Navigate to` localhost:3000/`
- Login and navigate the application
  - The constant components should be managing their state properly because they don't re-instantiate
  - The links on the left sidebar should be responding appropriately

#### Any background context you want to provide?
- The bug with the left sidebar is a result of constant components re-instantiating when navigating the application, thus unable to maintain and manage their own local state

#### What are the relevant pivotal tracker stories?
#160374729

#### Screenshots (if appropriate)

sep-10-2018 08-47-04 - sidebar

### CODE REVIEW

File: src/views/Base/index.jsx [hunk 1]
```diff
@@ -1,41 +1,12 @@
 import React, { Component } from 'react';
 import ConnectedLeftSideBar from '../../components/LeftSideBar/LeftSideBar';
```
Comment (Spectrumsun) [inline src/views/Base/index.jsx]: Please remove unused imports, NotificationPane, ConnectedSideDrawer, ConnectedLeftSideBar

File: src/views/Role/index.jsx [hunk 2]
```diff
@@ -75,34 +74,10 @@ export class Role extends Base {
     const { hideSideBar, selectedLink, hideNotificationPane } = this.state;
     let [hideClass, leftPaddingClass] = hideNotificationPane ? ['hide', ''] : ['', 'pd-left'];
```
Comment (Spectrumsun) [inline src/views/Role/index.jsx]: 'hideSideBar, selectedLink, hideNotificationPane' are not used in the component again. Please remove them from here and also from state.

**Comment by d3mola** [comment]
@johnmutuma5, I noticed that we have a new `Layout hoc` and some code from `Base` has gone into that, could you explain the rationale behind this?

File: src/views/Role/index.jsx (hunk 2 above)
Comment (johnmutuma5) [reply src/views/Role/index.jsx]: Great for this one too. Thanks @Spectrumsun

**Comment by d3mola** [comment]
@johnmutuma5, thanks for clarifying. PR looks good to me.

**Comment by johnmutuma5** [comment]
Thanks for the link @samuelkubai.

**Comment by johnmutuma5** [comment]
@samuelkubai, complete. Feedback put in place. Thanks.

**Comment by johnmutuma5** [comment]
@samuelkubai, components renamed.

**Comment by samuelkubai** [comment]
@johnmutuma5 please update from develop and ensure there are no warnings once you run the application.

**Comment by johnmutuma5** [comment]
@samuelkubai, branch updated.

**Pull request merged**